<!DOCTYPE html SYSTEM "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
     <head>
       <title>VarioML</title>
       <link href="_files/xmlns-style.css" rel="stylesheet" type="text/css"> 
  <link href="_files/style.css" rel="stylesheet" type="text/css"> 
  <link href="_files/style2.css" rel="stylesheet" type="text/css"> 
       <style type="text/css">table { border:2px; border-collapse:collapse; }</style>
     </head>
    <body>
    <H1>VarioML Documentation</H1>
    <p> ~ contents here ~ </p>
    <hr/>
     <h2 class='header'>lsdb (<a name='VmlLsdb'<em>VmlLsdb</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>id</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>schema_version</td><td>decimal</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>created</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>source</td><td><a href='#VmlSource'>VmlSource</a></td><td>zero or many</td><td></td></tr>
         <tr><td>individual</td><td><a href='#VmlIndividual'>VmlIndividual</a></td><td>one or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Single data submission batch element for individual data. The batch contains:</p>
<ul>
<li>id,uri - optional submission attributes<br /></li>
<li>created - optional timestamp when data submission batch was created<br /></li>
<li>source - information on source(s) of submission. Submitting source</li>
<li>list of individuals</li>
</ul>
<p><em>* Example:</em>*</p>
<lsdb xmlns="http://gen2phen.org/varioml/2.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:dc="example.org" xsi:noNamespaceSchemaLocation="lsdb.xsd"> <created>2000-01-12T12:13:14Z</created> <source id="AIREbase" version="1.2" uri="http://www.uta.fi/imt/bioinfo/AIREbase/"> <name>DatabaseXYZ</name> <url>http://www.uta.fi/imt/bioinfo/AIREbase/</url> <contact> <name>Matti Meikäläinen</name>
<address>
Institute for Molecular Medicine
</address>
        
<p></contact> <acknowledgement> <name>European Union</name> <grant_number accession="23230001"/> </acknowledgement> </source></p>
<pre><code>&lt;individual id=&quot;xyz&quot;&gt;
    &lt;gender code=&quot;2&quot;/&gt;
    &lt;variant&gt;
        &lt;ref_seq accession=&quot;xyz&quot;&gt;&lt;/ref_seq&gt;
        &lt;name scheme=&quot;HGVS&quot;&gt;c.1343GT&lt;/name&gt;
    &lt;/variant&gt;
&lt;/individual&gt;

&lt;individual id=&quot;abce&quot;&gt;
    &lt;gender code=&quot;2&quot;/&gt;
    &lt;variant&gt;
        &lt;ref_seq accession=&quot;xyz&quot;&gt;&lt;/ref_seq&gt;
        &lt;name scheme=&quot;HGVS&quot;&gt;c.1343GC&lt;/name&gt;
    &lt;/variant&gt;
&lt;/individual&gt;
</code></pre>
</lsdb>
 </div>
     <h2 class='header'>individual (<a name='VmlIndividual'<em>VmlIndividual</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>id</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>gender</td><td><a href='#VmlGender'>VmlGender</a></td><td>mandatory</td><td></td></tr>
         <tr><td>original_id</td><td><a href='#VmlOriginalId'>VmlOriginalId</a></td><td>optional</td><td></td></tr>
         <tr><td>organism</td><td><a href='#VmlOrganism'>VmlOrganism</a></td><td>optional</td><td></td></tr>
         <tr><td>phenotype</td><td><a href='#VmlPhenotype'>VmlPhenotype</a></td><td>zero or many</td><td></td></tr>
         <tr><td>population</td><td><a href='#VmlPopulation'>VmlPopulation</a></td><td>zero or many</td><td></td></tr>
         <tr><td>variant</td><td><a href='#VmlVariant'>VmlVariant</a></td><td>one or many</td><td></td></tr>
         <tr><td>source</td><td><a href='#VmlSource'>VmlSource</a></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td>creation_date</td><td><a href='#VmlCreationDate'>VmlCreationDate</a></td><td>optional</td><td></td></tr>
         <tr><td>modification_date</td><td><a href='#VmlModificationDate'>VmlModificationDate</a></td><td>optional</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Individual or patient</p>
<p>Example: <em>*</em></p>
<p><individual> <gender code="2" /> <phenotype term="Osteogenesis Imperfecta" > </phenotype> <variant type="cDNA"> <gene source="HGNC" accession="COL1A1" /> <ref_seq accession="NG_007400.1" /> <name scheme="HGVS">c.579delT</name> <genetic_origin term="paternal"> <evidence_code term="inferred"/><br /> </genetic_origin> <sharing_policy type="openAccess"> <embargo_end_date>2002-12-12</embargo_end_date> <use_permission accession="CC0" uri="http://creativecommons.org/publicdomain/zero/1.0/" term="Creative commons"/> </sharing_policy></p>
<pre><code>        &lt;comment&gt;
            &lt;text&gt;Variant inherited from affected father&lt;/text&gt;
        &lt;/comment&gt;

    &lt;/variant&gt;
&lt;/individual&gt;
</code></pre>
</div>
     <h2 class='header'>panel (<a name='VmlPanel'<em>VmlPanel</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>size</td><td>integer</td><td>optional</td><td></td></tr>
         <tr><td>id</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>individual</td><td><a href='#VmlIndividual'>VmlIndividual</a></td><td>zero or many</td><td></td></tr>
         <tr><td>frequency</td><td><a href='#VmlFrequency'>VmlFrequency</a></td><td>zero or many</td><td></td></tr>
         <tr><td>original_id</td><td><a href='#VmlOriginalId'>VmlOriginalId</a></td><td>optional</td><td></td></tr>
         <tr><td>organism</td><td><a href='#VmlOrganism'>VmlOrganism</a></td><td>optional</td><td></td></tr>
         <tr><td>phenotype</td><td><a href='#VmlPhenotype'>VmlPhenotype</a></td><td>zero or many</td><td></td></tr>
         <tr><td>population</td><td><a href='#VmlPopulation'>VmlPopulation</a></td><td>zero or many</td><td></td></tr>
         <tr><td>variant</td><td><a href='#VmlVariant'>VmlVariant</a></td><td>one or many</td><td></td></tr>
         <tr><td>source</td><td><a href='#VmlSource'>VmlSource</a></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td>creation_date</td><td><a href='#VmlCreationDate'>VmlCreationDate</a></td><td>optional</td><td></td></tr>
         <tr><td>modification_date</td><td><a href='#VmlModificationDate'>VmlModificationDate</a></td><td>optional</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Panel. Unspecified group of individuals. Has following information besides properties inherited from AbstractObservation target:</p>
<p><em>frequency - frequency of given variant or variant set</em>individuals - list of individuals beloning to the panel</p>
<p><em>* Example:</em>* Frequency for existence of two alleles. <panel> <frequency samples="224" type="allele"> <freq>0.763</freq> </frequency> <population type="group" term="El Salvadorans"/> <population type="ethnic" term="Mestizo"> <comment><text>Amerindian and European descent</text></comment> </population> <variant> <ref_seq accession="NM_000777"/> <name>g.6986AG</name><br /> </variant> <variant> <ref_seq accession="NM_000777"/> <name>g.1289TG</name><br /> </variant> </panel></p>
<p>Note also that also variant specific frequency can be given as usual:</p>
<pre><code>&lt;panel&gt;      
    &lt;population type=&quot;group&quot; term=&quot;Yorubans&quot;/&gt;
    &lt;population type=&quot;region&quot; term=&quot;Nigeria&quot;/&gt;
    &lt;variant&gt;
        &lt;ref_seq accession=&quot;NM_000777&quot;/&gt;
        &lt;name&gt;g.6986AG&lt;/name&gt;                                                
        &lt;frequency samples=&quot;50&quot; type=&quot;allele&quot;&gt; 
            &lt;freq&gt;0.06&lt;/freq&gt;
        &lt;/frequency&gt;
    &lt;/variant&gt;
&lt;/panel&gt;
</code></pre>
<p><strong>Usage of the format must be agreed on separately.</strong></p>
</div>
     <h2 class='header'>consent (<a name='VmlConsent'<em>VmlConsent</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>NOT USED</p>
<p>Database cross-reference for a consent information (not used currently)</p>
  </div>
     <h2 class='header'>gender (<a name='VmlGender'<em>VmlGender</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>code</td><td>string</td><td>mandatory</td><td>0 | 1 | 2 | 9</td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><a href='#VmlDescription'>VmlDescription</a></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Gender of individual. Gender code is stored in the attribute <em>* 'code'</em>* . Possible values are iso5218 codes: 0,1,2 and 9 ('unknown','male','female' and 'not applicable').</p>
<p>Description of gender (code: 9) can be stored into optional <em>* 'description'</em>* -sub-element, which is of type #type_LSDB.Evidence_ontology_term&quot; Evidence_ontology_term</p>
<p>.</p>
<p><em>* Example:</em>*</p>
<gender code='9'> <description term="XX-male"/> </gender>
</div>
     <h2 class='header'>description (<a name='VmlDescription'<em>VmlDescription</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>original_id (<a name='VmlOriginalId'<em>VmlOriginalId</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Identifier used by the submitter.</p>
  </div>
     <h2 class='header'>submitter_id (<a name='VmlSubmitterId'<em>VmlSubmitterId</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Randomized identifier of individual as used by submitter.</p>
  </div>
     <h2 class='header'>variant (<a name='VmlVariant'<em>VmlVariant</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>id</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>type</td><td>string</td><td>optional</td><td>DNA | cDNA | RNA | AA</td></tr>
         <tr><td>genotypic</td><td>string</td><td>optional</td><td>true</td></tr>
         <tr><td>subcellular_part</td><td>string</td><td>optional</td><td>nucleus | mitochondrial</td></tr>
         <tr><td>diploid_count</td><td>float</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>gene</td><td><a href='#VmlGene'>VmlGene</a></td><td>optional</td><td></td></tr>
         <tr><td>ref_seq</td><td><a href='#VmlRefSeq'>VmlRefSeq</a></td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td><a href='#VmlName'>VmlName</a></td><td>mandatory</td><td></td></tr>
         <tr><td>variant_type</td><td><a href='#VmlVariantType'>VmlVariantType</a></td><td>optional</td><td></td></tr>
         <tr><td>original_id</td><td><a href='#VmlOriginalId'>VmlOriginalId</a></td><td>optional</td><td></td></tr>
         <tr><td>exon</td><td><a href='#VmlExon'>VmlExon</a></td><td>zero or many</td><td></td></tr>
         <tr><td>sequence</td><td><a href='#VmlSequence'>VmlSequence</a></td><td>optional</td><td></td></tr>
         <tr><td>genotype</td><td><a href='#VmlGenotype'>VmlGenotype</a></td><td>optional</td><td></td></tr>
         <tr><td>consequence</td><td><a href='#VmlConsequence'>VmlConsequence</a></td><td>zero or many</td><td></td></tr>
         <tr><td>pathogenicity</td><td><a href='#VmlPathogenicity'>VmlPathogenicity</a></td><td>zero or many</td><td></td></tr>
         <tr><td>phenotype</td><td><a href='#VmlPhenotype'>VmlPhenotype</a></td><td>zero or many</td><td></td></tr>
         <tr><td>variant_detection</td><td><a href='#VmlVariantDetection'>VmlVariantDetection</a></td><td>optional</td><td></td></tr>
         <tr><td>restriction_site</td><td><a href='#VmlRestrictionSite'>VmlRestrictionSite</a></td><td>optional</td><td></td></tr>
         <tr><td>tissue_distribution</td><td><a href='#VmlTissueDistribution'>VmlTissueDistribution</a></td><td>optional</td><td></td></tr>
         <tr><td>genetic_origin</td><td><a href='#VmlGenetic_origin'>VmlGenetic_origin</a></td><td>zero or many</td><td></td></tr>
         <tr><td>frequency</td><td><a href='#VmlFrequency'>VmlFrequency</a></td><td>zero or many</td><td></td></tr>
         <tr><td>seq_changes</td><td><a href='#VmlConsequentVariants'>VmlConsequentVariants</a></td><td>optional</td><td></td></tr>
         <tr><td>aliases</td><td><a href='#VmlRelatedVariants'>VmlRelatedVariants</a></td><td>optional</td><td></td></tr>
         <tr><td>source</td><td><a href='#VmlSource'>VmlSource</a></td><td>optional</td><td></td></tr>
         <tr><td>haplotype_group</td><td><a href='#VmlHaplotypeGroup'>VmlHaplotypeGroup</a></td><td>optional</td><td></td></tr>
         <tr><td>location</td><td><a href='#VmlLocation'>VmlLocation</a></td><td>zero or many</td><td></td></tr>
         <tr><td>sharing_policy</td><td><a href='#VmlSharingPolicy'>VmlSharingPolicy</a></td><td>optional</td><td></td></tr>
         <tr><td>creation_date</td><td><a href='#VmlCreationDate'>VmlCreationDate</a></td><td>optional</td><td></td></tr>
         <tr><td>modification_date</td><td><a href='#VmlModificationDate'>VmlModificationDate</a></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Variant. Typically has the following information:</p>
<ul>
<li>type - DNA,cDNA,RNA,AA<br /></li>
<li>gene</li>
<li>ref_seq - reference sequence</li>
<li>name - name of variant, preferrably using the HGVS nomenclature<br /></li>
<li>consequence - consequence (list of consequences) of the variant using for example sequence or variation ontology</li>
<li>pathogenicity</li>
<li>phenotype - known phenotypic consequence</li>
<li>variant_detection - variant detection method</li>
<li>restriction_site - site of restriction enzyme</li>
<li>seq_changes - effects of variant on overlapping sequence levels/features</li>
<li><p>aliases - alternative representations of the variant</p>
<p><variant diploid_count='1'> <gene source='HGNC' accession='AIRE'/> <ref_seq source='IDRefSeq' accession='D0003' /> <name scheme='HGVS'>g.4707AT</name> <exon>e1</exon> <consequence term='A point mutation in the exon 1 altering initiation codon in HSR domain '/> <seq_changes> <variant type='RNA'> <ref_seq source='IDRefSeq' accession='C0003' /> <name scheme='HGVS'>r.1au</name> <consequence term='predicted to change start codon to position c.268, resulting in a frameshift of translation'/> <seq_changes> <variant type='AA'> <ref_seq source='UniProt' accession='O43918' /> <name scheme='HGVS'>p.Met1Leu</name> </variant> </seq_changes> </variant> </seq_changes> </variant></p></li>
</ul>
  </div>
     <h2 class='header'>seq_changes (<a name='VmlConsequentVariants'<em>VmlConsequentVariants</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>variant</td><td><a href='#VmlConsequentVariant'>VmlConsequentVariant</a></td><td>one or many</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Related sequence changes on overlapping sequence levels and features.</p>
<p>Element has recursive list of sequence variants of type RNA or AA. The Sub-variant element differ from the main variant element in a following way:</p>
<ul>
<li>type has only values RNA and AA<br /></li>
<li>ref_seq is optional</li>
<li>gene is missing</li>
</ul>
</div>
     <h2 class='header'>aliases (<a name='VmlRelatedVariants'<em>VmlRelatedVariants</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>variant</td><td><a href='#VmlVariant'>VmlVariant</a></td><td>one or many</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Aliases or alternative legacy definitions (names) of the variant.</p>
<p>Element has list of sequence variants (recursive data structure) which can be seen as aliases of the main variant. Typically the variants are annotated against different reference sequences, and/or their name is based on a different naming scheme.</p>
<p><em>* Example:</em>* *</p>
<pre><code>   &lt;variant&gt;
        &lt;ref_seq source=&quot;genbank&quot; accession=&quot;NG_001337.2&quot;&gt;&lt;/ref_seq&gt;
        &lt;name scheme=&quot;HGVS&quot;&gt;g.22048CG&lt;/name&gt;
        &lt;aliases&gt;
            &lt;variant&gt;
                &lt;ref_seq accession=&quot;NG_002335.2&quot;/&gt;
                &lt;name&gt;g.1400GC&lt;/name&gt;                    
            &lt;/variant&gt;
            &lt;variant&gt;
                &lt;ref_seq accession=&quot;NG_007072.2&quot;/&gt;
                &lt;name&gt;g.14034GC&lt;/name&gt;
            &lt;/variant&gt;
            &lt;variant&gt;
                &lt;ref_seq accession=&quot;NG_007088.1&quot;/&gt;
                &lt;name&gt;g.28259GC&lt;/name&gt;
            &lt;/variant&gt;                
        &lt;/aliases&gt;

        &lt;db_xref  source=&quot;dbsnp&quot; accession=&quot;rs13&quot;/&gt;
    &lt;/variant&gt;
</code></pre>
</div>
     <h2 class='header'>name (<a name='VmlName'<em>VmlName</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>scheme</td><td>string</td><td>optional</td><td></td></tr>
        </table>
     </div>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3><p>Name of variant</p>
<p>Element has following optional attributes: <em>* scheme</em>* which is naming scheme which should typically be HGVS.</p>
<p><em>* Example:</em>*</p>
<p><name type="DNA" scheme="HGVS"> c.755G </name></p>
</div>
     <h2 class='header'>sequence (<a name='VmlSequence'<em>VmlSequence</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>reference</td><td><em>string</em></td><td>mandatory</td><td></td></tr>
         <tr><td>variant</td><td><em>string</em></td><td>mandatory</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Reference and variant sequence strings. Accession number of reference sequence is given in ref_seq element (see the variant)</p>
<p><em>* Example:</em>* *</p>
<p><sequence> <reference>atttgatcgttc</reference> <variant>atttgGtcgttc</variant> </sequence></p>
</div>
     <h2 class='header'>reference</h2>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>variant</h2>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>genotype (<a name='VmlGenotype'<em>VmlGenotype</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>call</td><td><em>string</em></td><td>mandatory</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Genotype</p>
</div>
     <h2 class='header'>call</h2>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>ref_seq (<a name='VmlRefSeq'<em>VmlRefSeq</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Database cross reference for reference sequence.</p>
<p>Can include other database cross-references and annotations.</p>
<p><em>* Example:</em>* *</p>
<ref_seq source="genbank" accession="NG_007088.1"/>
</div>
     <h2 class='header'>protocol_id (<a name='VmlProtocolId'<em>VmlProtocolId</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Cross-reference to a protocol database.</p>
</div>
     <h2 class='header'>gene (<a name='VmlGene'<em>VmlGene</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Database cross reference where <strong>'source'</strong> is the name of the database or system (e.g. HUGO) and <strong>'accession'</strong> is the gene name (e.g.AGA).</p>
<p>HGNC names (symbols) should be used to name genes.</p>
<p>Can include other database cross-references and annotations.</p>
<p>Example source:</p>
<p>* HGNC</p>
<p>Example accession:</p>
<p>* BRACA2</p>
<p><em>* Examples:</em>*</p>
<gene source="HGNC" accession="BRACA2" />
<gene source="HGNC" accession="BRACA2"> <db_xref accession="600185" uri="http://www.ncbi.nlm.nih.gov/omim/600185" source="OMIM"/> <db_xref accession="BRACA2" source="HGNC" uri="http://www.genenames.org/data/hgnc_data.php?hgnc_id=1101"/> </gene>
</div>
     <h2 class='header'>modification_date (<a name='VmlModificationDate'<em>VmlModificationDate</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td></td><td><em>choice</em></td><td>optional</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Date when submitted information was modified.</p>
<p><em>* Example:</em>* *</p>
<p><modification_date>2001</modification_date> <modification_date>2001-09</modification_date> <modification_date>2001-09-10</modification_date></p>
</div>
     <h2 class='header'>creation_date (<a name='VmlCreationDate'<em>VmlCreationDate</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td></td><td><em>choice</em></td><td>optional</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Date when submitted information was created</p>
</div>
     <h2 class='header'>variant_type (<a name='VmlVariantType'<em>VmlVariantType</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>See types of sequence alteration in Sequence Ontology</p>
</div>
     <h2 class='header'>organism (<a name='VmlOrganism'<em>VmlOrganism</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Organism</p>
</div>
     <h2 class='header'>tissue_distribution (<a name='VmlTissueDistribution'<em>VmlTissueDistribution</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Tissue distribution</p>
<p>Controlled vocabulary terms for tissue disttribution: _</p>
<p>N.B: Standardization of this list is pending. See: http://www.gen2phen.org/wiki/lsdb-controlled-vocabulary-terms&quot; lsdb-controlled-vocabulary-terms</p>
<p>_</p>
<ul>
<li>constitutional</li>
<li>mosaic</li>
<li>mosaic in germline</li>
</ul>
<p><em>* Example:</em>* *</p>
<tissue_distribution term="mosaic"/>
</div>
     <h2 class='header'>sequence_region (<a name='VmlSequenceRegion'<em>VmlSequenceRegion</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Name of the sequence region (not used)</p>
</div>
     <h2 class='header'>restriction_site (<a name='VmlRestrictionSite'<em>VmlRestrictionSite</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Restriction enzyme site.</p>
</div>
     <h2 class='header'>exon (<a name='VmlExon'<em>VmlExon</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>transcript_ref</td><td>string</td><td>optional</td><td></td></tr>
        </table>
     </div>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3><p>Exon has optional attributes <strong>source</strong>,<strong>accession</strong> and <strong>transcript_ref</strong> which can be used for crossreference purposes. Exon number is given in text node</p>
<p><em>* Example:</em>*</p>
<p><exon>01</exon></p>
</div>
     <h2 class='header'>consequence (<a name='VmlConsequence'<em>VmlConsequence</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Consequences of mutation on sequence or structural levels.</p>
<p>_</p>
<p>N.B: Standardization of this list into the schema is still pending. See: http://www.gen2phen.org/wiki/lsdb-controlled-vocabulary-terms&quot; lsdb-controlled-vocabulary-terms</p>
<p>_</p>
<p><strong><em>Complex frameshift:</em>* Frameshift involving insertions and deletions</strong><em>Exon deletion:</em>* Deletion encompassing a whole exon or exons, frameshift status unknown <strong><em>Exon duplication:</em>* Duplication of one of more exons, frameshift status unknown</strong><em>Frameshift:</em>* Deletion or insertion causing reading frame shift <strong><em>In-frame deletion:</em>* Deletion of a whole codon or codons. Can include deletion of one or more exons</strong><em>In-frame duplication:</em>* A duplication that does not change the reading frame. Can include one or more exons <strong><em>In-frame insertion:</em>* An insertion of a whole codon or codons. Can include one or more exons</strong><em>Intronic variant:</em>* A variant in an intron which has not been shown to affect splicing <strong><em>Missense:</em>* Substitution resulting in a change to a different amino acid</strong><em>Nonsense:</em>* Substitution resulting in a change to a stop codon <strong><em>Out of frame deletion:</em>* Deletion of part of a codon or number of codons resulting in a frameshift. Can include one or more exons</strong><em>Out of frame duplication:</em>* A duplication that changes the reading frame. Can include one or more exons <strong><em>Out of frame insertion:</em>* An insertion of part of a codon or number of codons resulting in a frameshift. Can include one or more exons</strong><em>Silent:</em>* A nucleotide change that does not change the amino acid *<strong>Splice site variant:</strong> A mutation that affects splicing</p>
<p><em>* Example:</em>* *</p>
<consequence sourcels="SO" accession="1000071"
    term="mutation affecting splicing" />
</div>
     <h2 class='header'>genetic_origin (<a name='VmlGenetic_origin'<em>VmlGenetic_origin</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>source</td><td><a href='#VmlGeneticSource'>VmlGeneticSource</a></td><td>optional</td><td></td></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Genetic origin of variation. Has following information:</p>
<ul>
<li>term - ontology term for origin of variant (denovo, sporadic, inherited)<br /></li>
<li>source - parental origin of variation (paternal, maternal)</li>
<li>evidence code - evidence code for the observation</li>
<li>list of individuals</li>
</ul>
<p>* Genetic origin and source are evidence ontology terms. *</p>
<p>Controlled vocabulary terms for genetic origin: _</p>
<p>N.B: Standardization of this list is pending. See: http://www.gen2phen.org/wiki/lsdb-controlled-vocabulary-terms&quot; lsdb-controlled-vocabulary-terms</p>
<p>_</p>
<ul>
<li>inherited</li>
<li>sporadic</li>
<li>de novo</li>
</ul>
<p><em>* Example:</em>* *</p>
<genetic_origin term="inherited"> <source term="paternal"/> <evidence_code term="inferred"/><br /></genetic_origin>
</div>
     <h2 class='header'>source (<a name='VmlGeneticSource'<em>VmlGeneticSource</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>population</td><td><a href='#VmlPopulation'>VmlPopulation</a></td><td>zero or many</td><td></td></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Source of variant (paternal, maternal). See genetic_origin for more information.</p>
<p>Controlled vocabulary terms for genetic source: _ N.B: Standardization of this list is pending. See: http://www.gen2phen.org/wiki/lsdb-controlled-vocabulary-terms&quot; lsdb-controlled-vocabulary-terms</p>
<p>_</p>
<ul>
<li>paternal</li>
<li>maternal</li>
</ul>
</div>
     <h2 class='header'>location (<a name='VmlLocation'<em>VmlLocation</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>ref_seq</td><td><a href='#VmlRefSeq'>VmlRefSeq</a></td><td>mandatory</td><td></td></tr>
         <tr><td>chr</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>start</td><td><em>string</em></td><td>mandatory</td><td></td></tr>
         <tr><td>end</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Location of variant on a reference sequence/genomic build</p>
<p><em>* Example:</em>* *</p>
<p><location> <ref_seq source="embl" accession="ABC1023345"/> <start>111111</start> <end>111112</end> </location></p>
<p><location> <ref_seq accession="build36"/> <chr>21</chr> <start>111111</start> <end>111112</end> </location></p>
 
</div>
     <h2 class='header'>start</h2>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>sharing_policy (<a name='VmlSharingPolicy'<em>VmlSharingPolicy</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>type</td><td>string</td><td>mandatory</td><td>closedAccess | embargoedAccess | restrictedAccess | openAccess</td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>embargo_end_date</td><td><a href='#VmlEmbargo_end_date'>VmlEmbargo_end_date</a></td><td>optional</td><td></td></tr>
         <tr><td>use_permission</td><td><a href='#VmlUsePermission'>VmlUsePermission</a></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Data sharing policy. Following access policy types are used: closedAccess,embargoedAccess,restrictedAccess and openAccess * Adopted from OpenAIRE : &lt;a href='http://wiki.surffoundation.nl/display/standards/info-eu-repo/#info-eu-repo-AccessRights' specification</p>
<p><em>* Example:</em>* *</p>
<pre><code>    &lt;sharing_policy type=&quot;openAccess&quot;&gt;
        &lt;embargo_end_date&gt;2002-12-12&lt;/embargo_end_date&gt;
        &lt;use_permission accession=&quot;CC0&quot; uri=&quot;http://creativecommons.org/publicdomain/zero/1.0/&quot; term=&quot;Creative commons&quot;/&gt;
    &lt;/sharing_policy&gt;
</code></pre>
</div>
     <h2 class='header'>embargo_end_date (<a name='VmlEmbargo_end_date'<em>VmlEmbargo_end_date</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>is_undefined</td><td>string</td><td>optional</td><td>true</td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Embargo end date</p>
</div>
     <h2 class='header'>use_permission (<a name='VmlUsePermission'<em>VmlUsePermission</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Licensing information.</p>
<p><em>* Example:</em>* *</p>
<use_permission accession="CC0" uri="http://creativecommons.org/publicdomain/zero/1.0/" term="Creative commons"/>
</div>
     <h2 class='header'>pathogenicity (<a name='VmlPathogenicity'<em>VmlPathogenicity</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>scope</td><td>string</td><td>optional</td><td>individual | family | population</td></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Pathogenicity of variant.</p>
<p>Pathogenicity has 7 possible values:</p>
<p>* Non-pathogenic * Not known * Pathogenic * Probably not pathogenic * Probably pathogenic * Causative * Probably causative * Unclassified</p>
<p>Note: implementation of fixed vocabulary for this element is still pending. See: http://www.gen2phen.org/wiki/lsdb-controlled-vocabulary-terms&quot; lsdb-controlled-vocabulary-terms</p>
<p>Causative are added as workaround for cases where mutation do not lead to a disease phenotype (e.g. eye color etc.)</p>
<p>Causative is for cases where variant is known to have impact on non disease phenotype.</p>
<p>Pathogenicity has new attributes: phenotype, scope (individual, family, population) and score</p>
</div>
     <h2 class='header'>haplotype_group (<a name='VmlHaplotypeGroup'<em>VmlHaplotypeGroup</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>id</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Haplotype group id. Can be used for marking up alleles which are on same phase.</p>
</div>
     <h2 class='header'>variant_detection (<a name='VmlVariantDetection'<em>VmlVariantDetection</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>template</td><td>string</td><td>mandatory</td><td>DNA | RNA | cDNA | AA</td></tr>
         <tr><td>technique</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Techniques used to detect the variant.</p>
<p>Variant detection method. Element has attributes:</p>
<p>* sequence template, which can be either DNA, RNA, cDNA or AA * detection technique</p>
<p>Optional element protocol_id (of type DbXRef) has reference to actual detection protocol.</p>
<p>Controlled vocabulary terms for detection techniques: _</p>
<p>N.B: Standardization of this list is pending. See: http://www.gen2phen.org/wiki/lsdb-controlled-vocabulary-terms&quot; lsdb-controlled-vocabulary-terms</p>
<p>_</p>
<p><strong><em>Not Known:</em>* The information has not been recorded or provided</strong> <strong><em>Not Specified:</em>* Test information cannot be determined</strong> <strong><em>ARMS</em>*</strong><em>BESS:</em>* Base Excision Sequence Scanning<em>* <strong><em>CF20:</em>* CF Common Mutation Test</strong> <strong><em>CF29:</em>* Analysis of 29 mutations using the Elucigene CF29 kit</strong> <strong><em>CMC:</em>* Chemical Mismatch Cleavage</strong> <strong><em>CSCE:</em>* Conformation sensitive capillary electrophoresis</strong> <strong><em>DGGE:</em>* Denaturing-Gradient Gel-Electrophoresis</strong> <strong><em>DHPLC:</em>* Denaturing High-Performance Liquid Chromatography</strong> <strong><em>DOVAM:</em>* Detection Of Virtually All Mutations (SSCA variant)</strong> <strong><em>DSCA:</em>* Double-Strand DNA Conformation Analysis</strong> <strong><em>HD:</em>* HeteroDuplex analysis</strong> <strong><em>IHC:</em>* Immuno-Histo-Chemistry</strong> <strong><em>Loss of heterozygosity analysis</em>*</strong></em>mPCR:<em>* multiplex PCR</em>* <strong><em>Meta-PCR</em>*</strong><em>MS-PCR:</em>* Mutagenically separated PCR<em>* <strong><em>MAPH:</em>* Multiplex Amplifiable Probe Hybridisation</strong> <strong><em>MLPA:</em>* Multiplex Ligation-dependent Probe Amplification</strong> <strong><em>NGS:</em>* Next Generation Sequencing</strong> <strong><em>PAGE:</em>* Poly-Acrylamide Gel-Electrophoresis</strong> <strong><em>PCR:</em>* Polymerase Chain Reaction</strong> <strong><em>PTT:</em>* Protein Truncation Test</strong> <strong><em>RT-PCR:</em>* Reverse Transcription and PCR</strong> <strong><em>RNA:</em>* RNA work performed</strong> <strong><em>SEQ:</em>* SEQuencing</strong> <strong><em>SNPlex:</em>* The SNPlex™ Genotyping System from ABI</strong> <strong><em>Southern:</em>* Southern Blotting</strong> <strong><em>SSCA:</em>* Single-Strand DNA Conformation Analysis (SSCP)</strong> <strong><em>SSCP/Heteroduplex</em>*</strong></em>Western:<em>* Western Blotting</em>*</p>
<p><em>* Example:</em>*</p>
<variant_detection template="DNA" technique="DHPLC"/>
</div>
     <h2 class='header'>db_xref (<a name='VmlDbXRef'<em>VmlDbXRef</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>mandatory</td><td></td></tr>
         <tr><td>name</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Abstract data type for database cross references.</p>
<p>Can include other database cross-references and annotations.</p>
<p><em>* Examples:</em>*</p>
<db_xref source="pubmed" accession="16965330"/>
</div>
     <h2 class='header'>phenotype (<a name='VmlPhenotype'<em>VmlPhenotype</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>inheritance_pattern</td><td><a href='#VmlInheritancePattern'>VmlInheritancePattern</a></td><td>optional</td><td></td></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Phenotype of individual. Phenotype is evidence ontology term (see evidence_ontology)</p>
<p><em>* Example:</em>* *</p>
<phenotype term='Autoimmune polyglandular syndrome type 1'/>
<p><phenotype term='Autoimmune polyglandular syndrome type 1' accession='240300' source='OMIM'> <evidence_code term="confirmed" source="vario"/> <protocol_id accession="observation_method_xyz" source="protocol_database"/> <db_xref accession="12345" source="pubmed"/> <comment term='symptom'><text>Hypoparathyroidism</text></comment> <comment term='symptom'><text>Addison's disease</text></comment> <comment term='symptom'><text>Mucocutaneous candidiasis</text></comment> <comment term='symptom'><text>Malabsorption</text></comment> <comment term='symptom'><text>Chronic active hepatitis</text></comment> </phenotype></p>
</div>
     <h2 class='header'>inheritance_pattern (<a name='VmlInheritancePattern'<em>VmlInheritancePattern</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Inheritance pattern of the phenotype. Possible values for the term:</p>
<ul>
<li>familial<br /></li>
<li>familial, consanguineous parents<br /></li>
<li>familial, autosomal dominant</li>
<li>familial, autosomal recessive<br /></li>
<li>familial, Xlinked<br /></li>
<li>sporadic<br /></li>
<li>sporadic, consanguineous parents<br /></li>
<li>sporadic, consanguineous parents (1st degree)<br /></li>
<li>sporadic, consanguineous parents (2nd degree)<br /></li>
<li>sporadic, consanguineous parents (3rd degree)<br /></li>
<li>sporadic, nonconsanguineous parents<br /></li>
<li>sporadic, consanguinity parents?<br /></li>
<li>sporadic? (parents not tested)</li>
</ul>
<p>See: http://www.gen2phen.org/wiki/lsdb-controlled-vocabulary-terms&quot; lsdb-controlled-vocabulary-terms</p>
</div>
     <h2 class='header'>frequency (<a name='VmlFrequency'<em>VmlFrequency</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>samples</td><td>int</td><td>optional</td><td></td></tr>
         <tr><td>type</td><td>string</td><td>optional</td><td>allele | carrier</td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>population</td><td><a href='#VmlPopulation'>VmlPopulation</a></td><td>optional</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>optional</td><td>freq|counts|category</td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Actual frequency is stored in a subelement depending on value type. This type can be: decimal number, number of cases, or categorized value.</p>
<p>Frequency has the optional attribute 'samples', which can hold information on sample size. Typically this is the number of chromosomes. The calculation method type should be cross-referenced using protocol_id.</p>
<p>Frequency type can be either &quot;allele&quot; or &quot;carrier&quot;, with optional subelements:</p>
<p>* population * Evidence_ontology_term-type * evidence code * protocol id * comment</p>
<p><em>* Examples:</em>*</p>
<p><frequency samples="1000" > <population type="nationality" term="Finnish"/> <freq>0.2</freq> </frequency></p>
<p><frequency><freq>0.1</freq></frequency></p>
<p><frequency samples="1000"> <cases>11</cases> </frequency></p>
<frequency> <population type="region" term="Sweden"/> <category term="exist"/> </frequency>
<p><frequency samples="1111"> <population type="ethnic" term="Caucasian"/> <freq>0.12</freq> <evidence_code source="sampling_ontology" accession="abcd" term="random"/> </frequency></p>
</div>
     <h2 class='header'>freq</h2>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>counts</h2>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>category (<a name='VmlOntologyTerm'<em>VmlOntologyTerm</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>population (<a name='VmlPopulation'<em>VmlPopulation</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>type</td><td>string</td><td>mandatory</td><td>region | nationality | ethnic | race | organism | group | primary_language | language_family | other | unknown</td></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Information related to population group the individual belongs to.</p>
<p>Population element has the &quot;type&quot; attribute which indicates the selection criteria of the population, using the values:</p>
<p>* ethnic * region * race * other * unknown</p>
<p>The attribute is informative: the actual type of population should be defined in the source ontology (if it is used).</p>
<p><em>* Example:</em>*</p>
<population type="ethnic" term="Caucasian"/>
<population type="ethnic" term="Caucasian"> <evidence_code source="example_ontology_term" accession="abcd" term="ethnicity of mother"/> </population>
<population type="ethnic" term="Asian" > <evidence_code source="example_ontology_term" accession="abcd" term="ethnicity of father"/> </population>
<population type="region" term="Ahvenanmaa" source="ISO 3166"  accession="ISO 3166-2:AX"> <evidence_code term="place of birth"/> </population>
</div>
     <h2 class='header'>evidence_code (<a name='VmlEvidenceCode'<em>VmlEvidenceCode</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>mandatory</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>description</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Evidence code is an ontology term (i.e. has optional source and accession attributes), which provides information related to the validity or the quality of observation.</p>
<p><em>* Examples:</em>* *</p>
<evidence_code term="inferred"/>
<evidence_code term="confirmed" source="vario"/>
<p>dbSNP validation should be used for mutations Not validated (accession=0), Multiple reporting (1), With frequency (2),Both frequency (3), Submitter validation (4)</p>
</div>
     <h2 class='header'>comment (<a name='VmlComment'<em>VmlComment</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>source</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>accession</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>term</td><td>string</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>text</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>evidence_code</td><td><a href='#VmlEvidenceCode'>VmlEvidenceCode</a></td><td>zero or many</td><td></td></tr>
         <tr><td>protocol_id</td><td><a href='#VmlProtocolId'>VmlProtocolId</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Comments are of type Evidence_ontology. The term 'attribute' is optional. See under description of &quot;Evidence_ontology_term data type.</p>
<p>Comments can contain other comments.</p>
<p><em>* Examples:</em>*</p>
<p><comment> <text>simple comment</text> </comment></p>
<p><comment source="example_ontology" term="example term"> <text>my defined comment</text> </comment></p>
<p><comment> <text>whatever</text> <comment> <text>my structured comment</text> </comment> </comment></p>
<p><comment> <text>whatever</text> <comment> <text>my complex structured comment</text> <evidence_code term="something"/> <db_xref accession="xyz"/> </comment> </comment></p>
<p><phenotype term='Autoimmune polyglandular syndrome type 1'> <comment term='symptom'><text>Hypoparathyroidism</text></comment> <comment term='symptom'><text>Addison's disease</text></comment> <comment term='symptom'><text>Mucocutaneous candidiasis</text></comment> <comment term='symptom'><text>Malabsorption</text></comment> <comment term='symptom'><text>Chronic active hepatitis</text></comment> </phenotype></p>
</div>
     <h2 class='header'>source (<a name='VmlSource'<em>VmlSource</em></a>)</h2>
     <div class='attributes'>
       <h3>Attributes</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>value</th></tr>
         <tr><td>id</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>uri</td><td>anyURI</td><td>optional</td><td></td></tr>
         <tr><td>version</td><td>string</td><td>optional</td><td></td></tr>
         <tr><td>date</td><td>date</td><td>optional</td><td></td></tr>
        </table>
     </div>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>name</td><td><em>string</em></td><td>mandatory</td><td></td></tr>
         <tr><td>url</td><td><em>string</em></td><td>zero or many</td><td></td></tr>
         <tr><td>contact</td><td><a href='#VmlContact'>VmlContact</a></td><td>zero or many</td><td></td></tr>
         <tr><td>acknowledgement</td><td><a href='#VmlAcknowledgement'>VmlAcknowledgement</a></td><td>zero or many</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Contains information on data source. Typically specifies the source database, although the system, organization, or institution from which data is exported can also be used.</p>
<p>Element is re-used from the http://www.lrg-sequence.org&quot;LRG specification. In the LSDB XML format there can be more than one source, so that information on original data providers can be included.</p>
<p><em>* Example:</em>*</p>
<source id="dmudb"><br /> <name>DMuDB, NGRL, Manchester</name> <url>http://www.ngrl.org.uk/Manchester/projects/informatics/dmudb</url><br /> <contact> <name>Glen Dobson</name>
<address>
NGRL, Genetic Medicine, St Mary's Hospital, Oxford Road, Manchester M13 9WL
</address>
        
<p><phone>(+44) 0000 000 0000</phone> <fax>(+44) 0000 000 0000</fax> <email>example@nnn.nn.nnn</email> </contact> </source></p>
</div>
     <h2 class='header'>name</h2>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>contact (<a name='VmlContact'<em>VmlContact</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>name</td><td><em>string</em></td><td>mandatory</td><td></td></tr>
         <tr><td>url</td><td><em>string</em></td><td>zero or many</td><td></td></tr>
         <tr><td>address</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>phone</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>fax</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>email</td><td><em>string</em></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>name</h2>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3></div>
     <h2 class='header'>acknowledgement (<a name='VmlAcknowledgement'<em>VmlAcknowledgement</em></a>)</h2>
     <div class='elements'>
       <h3>Elements</h3>
       <table border='1'>
         <tr><th>name</th><th>type</th><th>cardinality</th><th>choice group</th></tr>
         <tr><td>name</td><td><em>string</em></td><td>mandatory</td><td></td></tr>
         <tr><td>grant_number</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>optional</td><td></td></tr>
         <tr><td>db_xref</td><td><a href='#VmlDbXRef'>VmlDbXRef</a></td><td>zero or many</td><td></td></tr>
         <tr><td>comment</td><td><a href='#VmlComment'>VmlComment</a></td><td>zero or many</td><td></td></tr>
         <tr><td></td><td><em>choice</em></td><td>zero or many</td><td></td></tr>
        </table>
     </div>
     </p><div class='doc'> <h3>Documentation</h3><p>Acknowledgement.</p>
  </div>
     <h2 class='header'>name</h2>
</p><d class='text'>text node </d>
     </p><div class='doc'> <h3>Documentation</h3></div>
    </body>
</html>)
